keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryPromotionRepository.kt

```diff
@@ -0,0 +1,59 @@
+package com.netflix.spinnaker.keel.persistence.memory
+
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.persistence.PromotionRepository
+
+class InMemoryPromotionRepository : PromotionRepository {
+
+  private val approvedVersions = mutableMapOf<Triple<DeliveryArtifact, DeliveryConfig, String>, String>()
+  private val deployedVersions = mutableMapOf<Triple<DeliveryArtifact, DeliveryConfig, String>, MutableList<String>>()
```

Comment:
is there a benefit to having this as a `Triple` as opposed to making a data class to hold these three related things together?

Reply:
It also seems like a lot of data to store (a duplicate of the whole delivery config). Maybe we need to generate ids from delivery configs and just store that? Can be a future optimization.

Reply:
Right now they don't have ids. I've been thinking about whether we want to add that but I figured it could be a future optimization. There are other issues; do we want `DeliveryConfig`, `Environment` and `Resource` to be all extensions of the same thing? Maybe they all need `metadata`?

Reply:
hmm okay. seems like a good thing to come back to in the future if it's a problem.

Reply:
Also, this is just the in-memory repository so it doesn't need to be super optimal.